xdsl/dialects/llvm.py

```diff
@@ -641,22 +649,22 @@ class ICmpOp(IRDLOperation):
 
     lhs = operand_def(T)
     rhs = operand_def(T)
-    res = result_def(T)
-    predicate = prop_def(ICmpPredicateAttr)
+    res = result_def(i1)
+    predicate = prop_def(IntegerAttr[i64])
 
     traits = frozenset([NoMemoryEffect()])
 
     def __init__(
         self,
         lhs: SSAValue,
         rhs: SSAValue,
-        predicate: ICmpPredicateAttr,
+        predicate: IntegerAttr[IntegerType],
```

Comment:
Should this not also be `IntegerAttr[i64]`?

Reply:
no, I get the error `Variable not allowed in type expression` if I do that
Doing `IntegerAttr[IntegerType(64)]` gives the error `Call expression not allowed in type expression`

Reply:
Yep that's because the Python type system cannot do the things we do at runtime for verification.